java-checks/src/main/java/org/sonar/java/checks/security/FilePermissionsCheck.java

```diff
@@ -112,10 +122,43 @@ private static boolean sensitivePermissionsAsString(ExpressionTree arg0) {
       .isPresent();
   }
 
-  private static boolean sensitiveChmodCommand(ExpressionTree arg0) {
-    return arg0.asConstant(String.class)
-      .filter(cmd -> cmd.trim().startsWith("chmod"))
-      .filter(cmd -> CHMOD_OCTAL_PATTERN.matcher(cmd).find() || SIMPLIFIED_CHMOD_OTHER_PATTERN.matcher(cmd).find())
+  private void checkExecSingleStringArgument(ExpressionTree arg0) {
+    if (chmodCommand(arg0).filter(FilePermissionsCheck::isSensisitiveChmodMode).isPresent()) {
+      reportIssue(arg0, ISSUE_MESSAGE);
+    }
+  }
+
+  private void checkExecStringArrayArgument(NewArrayTree newArrayTree) {
+    List<ExpressionTree> initializers = newArrayTree.initializers();
+    if (initializers.size() < 3) {
+      // malformed
+      return;
+    }
+    if (!chmodCommand(initializers.get(0)).isPresent()) {
+      return;
+    }
+    ExpressionTree modeArg = initializers.get(1);
+    if (initializers.size() > 3 && isRecursiveArgument(modeArg)) {
```

Comment:
`--recursive` is just one of the possible options that can be pass to `chmod`, but there are others and multiple options can be passed.
No need to know at which index is the `mode` argument, we can safely test `isSensisitiveChmodMode` on all the `initializers` list (skipping first).